src/player.rs

```diff
@@ -51,17 +51,14 @@ pub async fn fetch_update(state: Arc<GlobalState>) -> Result<(), FetchUpdateStat
     let mut player_id: u32 = u32::from_str_radix(player_id_str, 16).unwrap();
 
     let mut current_player_info = global_state.player_info.lock().await;
-    let current_player_id = current_player_info.player_id;
+    let mut current_player_id = current_player_info.player_id;
```

Comment:
I think you meant to make `player_id` not mutable instead of adding `mut` to `current_player_id` 

```suggestion
    let player_id: u32 = u32::from_str_radix(player_id_str, 16).unwrap();
```

```suggestion
    let current_player_id = current_player_info.player_id;
```